docs/design/hts_address_tx_receipt.md

```diff
@@ -0,0 +1,62 @@
+# HTS Token Creation via SystemContract Receipt Enhancement
+
+## Purpose
+The purpose of this enhancement is to improve the interoperability between Hedera Token Service (HTS) and Ethereum-compatible tools, like ethers, by ensuring access to the necessary information about newly created tokens, in this case token address. This change specifically addresses the gap in Web3 tooling support when creating tokens through HTS precompile contracts, making user experience smoother.
+
+
+## Problem Statement
+When users create tokens (fungible and non-fungible) by directly calling the 0x167 contract there is no way for them to get the evm address of the newly created token via standard EVM tooling e.g ethers. This is because the transaction receipt's contractAddress field is not being populated with the HTS token address, since essentially the operations is just a contract call, but not a contract create (deploy).
+
+## Current Behaviour
+
+Currently, when creating a token via HTS the address of the system contract is returned as a contractAddress in the transaction receipt.
+
+## Proposed Solution
+
+### Technical Details
```

Comment:
Is it possible to mention somewhere the endpoints are being modified for this logic?

Reply:
Well, this affects the getTransactionReceipt endpoint only
As it is mentioned in a few places that we want to return it there?
Could you suggest a specific place?

Reply:
Ahh okay I see. I would suggest creating a dedicated section to clarify that the feature focuses on specific endpoint. As when I read through the doc, I was wondering what endpoints are being focused. So perhaps we could add a section and title it "Affected Endpoints," "Impacted Endpoints," "Targeted Endpoints," or something like that.